app/uk/gov/hmrc/gform/gform/DeclarationController.scala

```diff
@@ -215,7 +228,8 @@ class DeclarationController(
   }
 
   private def processValidation[U <: SectionSelectorType: SectionSelector](
-    validationResult: ValidationResult,
+    validationFormModelResult: ValidationResult,
+    validationDeclarationSectionResult: ValidationResult,
```

Comment:
This feels unnecessary. Why to pass two `ValidationResult`? (note. I understand why you do it, given the approach you've taken.)
Suggestion: Original ` validationResult: ValidationResult,` contains all the information needed to detect where the error occurred. Whole implementation can be done solely in `processInvalid` without need to change any signature and existing logic.

Hint if you improve `ValidationResult` to give us back `FormComponentId` with error, something like this (incomplete):
```
val errorsFields: List[FormComponentId] = lookup.collect {
  case (fcId, t: FieldError)       => fcId
  case (fcId, t: FieldGlobalError) => fcId
}.toList
```
you can then easily check if error happened on Declaration section or not.